File: week2/Eren-Taskiran/739-daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temps) {
        int[] results = new int[temps.length];
        Stack<Pair<Integer, Integer>> stack = new Stack<>();

        for (int i = 0; i < temps.length; i++) {
            while (!stack.isEmpty() && temps[i] > stack.peek().getKey()) {
                Pair<Integer, Integer> curr = stack.pop();
                results[curr.getValue()] = i - curr.getValue();
            }
            stack.push(new Pair<>(temps[i], i));
        }
        return results;
    }
}

File: week2/furkanunsalan/739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temps):
        results = [0] * len(temps)
        stack = []
        for i, temp in enumerate(temps):
            while stack and temps[stack[-1]] < temp:
                index = stack.pop()
                results[index] = i - index
            stack.append(i)

        return results